chrome/content/zotero/fileInterface.js

```diff
@@ -122,33 +122,47 @@ Zotero_File_Exporter.prototype.save = async function () {
 		translation.setLibraryID(this.libraryID);
 	}
 	
-	translation.setLocation(Zotero.File.pathToFile(fp.file));
+	function _exportDone(obj, worked) {
+		// Close the items exported indicator
+		Zotero_File_Interface.Progress.close();
+
+		if (!worked) {
+			Zotero.alert(
+				null,
+				Zotero.getString('general.error'),
+				Zotero.getString('fileInterface.exportError')
+			);
+			return;
+		}
+
+		// For Note Markdown translator replace zotero:// URI scheme,
+		// if the current app is not Zotero
+		if (io.selectedTranslator.translatorID == '154c2785-ec83-4c27-8a8a-d27b3a2eded1'
+			&& ZOTERO_CONFIG.ID != 'zotero') {
+			let text = obj.string;
+			text = text.replace(/zotero:\/\//g, ZOTERO_CONFIG.ID + '://');
+			Zotero.File.putContentsAsync(fp.file, text);
```

Comment:
Hmm, is there not a way to do this without two separate code paths for saving the file (`setLocation()` and `putContentsAsync`)?

If not, it seems like `_exportDone()` should be `async`, this should be `await`, and this should run before we close the progress dialog if `worked` is true, since the save could theoretically take some time.

But it would be better not to have to have to handle this in two different places.

Reply:
Oh, and actually, is the progress window already saying the export is done by this point, and we're just closing it here? So all the more reason to do this substitution during the actual processing if we can.

Reply:
It should be fixed now.

> Hmm, is there not a way to do this without two separate code paths for saving the file (`setLocation()` and `putContentsAsync`)?

`setLocation` uses streaming (if I'm right) which is useful for other translators, and for the replacement we need to take over and replace the whole output, therefore, no, we probably can't avoid two code paths.

> Oh, and actually, is the progress window already saying the export is done by this point, and we're just closing it here?

Popup closing was already here before. So, I guess, the export should be done at that point. Although when using RDF translator and including PDFs and annotations, the dialog closes much earlier that it actually finishes (pdf-worker is doing export) and we aren't waiting for [that](https://github.com/zotero/zotero/blob/3f37786ffd4fc1c977d4b16eb841f6619680372a/chrome/content/zotero/xpcom/translation/translate_item.js#L1223) I guess.

Reply:
Just one thing - I still miss a way to export multiple individual notes.

Reply:
OK, I have a not-totally-hacky way to do this. Will commit in a bit.